integreat_cms/cms/models/contact/contact.py

```diff
@@ -14,18 +14,38 @@ class Contact(AbstractBaseModel):
     """
 
     title = models.CharField(max_length=200, verbose_name=_("title"))
-    name = models.CharField(max_length=200, verbose_name=_("name"))
-    poi = models.ForeignKey(
-        POI, on_delete=models.PROTECT, verbose_name=_("POI"), related_name="contacts"
+    name = models.CharField(max_length=200, blank=True, verbose_name=_("name"))
+    location = models.ForeignKey(
+        POI,
+        on_delete=models.PROTECT,
+        verbose_name=_("location"),
+        related_name="contacts",
     )
     email = models.EmailField(
         blank=True,
         verbose_name=_("email address"),
     )
+    email_from_location = models.BooleanField(
+        default=True,
+        verbose_name=_("E-mail from location"),
+        help_text=_("Will be taken over from the location."),
+    )
     phone_number = models.CharField(
-        max_length=250, blank=True, verbose_name=_("phone number")
+        max_length=250,
+        blank=True,
+        verbose_name=_("phone number"),
+    )
+    phone_number_from_location = models.BooleanField(
+        default=True,
+        verbose_name=_("phone number from location"),
+        help_text=_("Will be taken over from the location."),
     )
     website = models.URLField(blank=True, max_length=250, verbose_name=_("website"))
+    website_from_location = models.BooleanField(
+        default=True,
+        verbose_name=_("website from location"),
+        help_text=_("Will be taken over from the location."),
+    )
```

Comment:
I'm not sure if it is then self explaining enough for users so I prefer to set a checkbox (or something similar). Also having those fields is probably helpful for #3033

Reply:
Oh I totally agree on the user interface, I didn't mean to change that at all. I'm not sure what you mean with the fields being helpful – from the perspective of the model it should be just as simple to query for a field being `NULL` as for a bool being true or false

Reply:
An E-mail, phone number or website is given but the corresponding check box is checked and the given value does not match with that of the POI -such case must not happen in future, becuase 1) the field will be automatically filled with the data of the selected POI and not editable when the check box is checked (this is implemented in this PR) and 2) those data will syncrhonize when they must be taken from the POI #3033

And I'm not fun of adding an extra complexity or agreement. Even if we document it, we will fail to be careful enough to it someday, and differencisting between `Null` and `""` is prone to bugs (in my opinion)

Reply:
I am also not a fan of having these `*_from_location` fields, but I also agree with Mizuki that the distinction between NULL and "" may be error prone. In my opinion, we should not have two models with redundant data in them. So I proposed an alternative to #3033: https://github.com/digitalfabrik/integreat-cms/issues/3033#issuecomment-2360829708

In that case, we would not need the `*_from_location` fields at all.

@MizukiTemma @PeterNerlich What are your opinions on that?

Reply:
No, what I meant was that we allow users to set anything they want in contacts, but prevent users from editing the email, website, phone number fields of pois. So your example scenario would still be no problem, as users could create multiple contacts with different phone numbers for the same poi

Reply:
>prevent users from editing the email, website, phone number fields of pois.

🤔 ??

The details of POIs will not be changed from the contact form in the current implementation too. Users can only create a brand new POI (in case the wished one does not exist yet). E-mail, phone numebr and website URL given in the contact form will not be transfered to the POI.